pages/search/index.js

```diff
@@ -1,48 +1,308 @@
 import styled from "styled-components";
 import VerticalGameList from "@/components/VerticalGameList";
 
+import Fuse from "fuse.js";
+
 import { useState } from "react";
 import useSWR from "swr";
+import { FaTimes } from "react-icons/fa";
+
+const INITIAL_DATA = {
+  name: "",
+  categories: "",
+  rating: "",
+  minAge: "",
+  players: "",
+  yearpublished: "",
+  playtime: "",
+};
 
-export default function SearchPage({ isFavorite, toggleFavorite, showModal }) {
+export default function SearchPage({ isFavorite, toggleFavorite }) {
   const { data } = useSWR("./api/games");
-  const [searchInput, setSearchInput] = useState("");
+  const [searchQuery, setSearchQuery] = useState("");
+
+  const [showFilters, setShowFilters] = useState(false);
+  const [filteredResults, setFilteredResults] = useState([]);
+
+  const [formData, setFormData] = useState(INITIAL_DATA);
```

Comment:
this state is a bit missleading, please rename it to something like "searchFormData"